toolkit/tools/pkg/imagecustomizerlib/liveosisobuilder.go

```diff
@@ -1289,16 +1471,86 @@ func (b *LiveOSIsoBuilder) createImageFromUnchangedOS(baseConfigPath string, iso
 		return fmt.Errorf("failed to convert iso configuration to isomaker configuration format:\n%w", err)
 	}
 
-	err = b.updateGrubCfg(b.artifacts.savedKernelArgsFilePath, b.artifacts.grubCfgPath, extraCommandLine)
+	pxeIsoImageUrl := ""
+	if pxeConfig != nil {
+		pxeIsoImageUrl = pxeConfig.IsoImageUrl
+	}
+
+	err = b.updateGrubCfg(b.artifacts.savedConfigsFilePath, b.artifacts.isoGrubCfgPath, extraCommandLine, b.artifacts.pxeGrubCfgPath, pxeIsoImageUrl, outputImageBase)
 	if err != nil {
 		return fmt.Errorf("failed to update grub.cfg:\n%w", err)
 	}
 
-	err = b.createIsoImage(additionalIsoFiles, outputImageDir, outputImageBase)
+	isoImagePath, err := b.createIsoImage(additionalIsoFiles, outputImageDir, outputImageBase)
 	if err != nil {
 		return fmt.Errorf("failed to create iso image:\n%w", err)
 	}
 
+	err = populatePXEArtifactsDir(isoImagePath, b.workingDirs.isoBuildDir, outputPXEArtifactsDir, outputImageBase)
```

Comment:
`populatePXEArtifactsDir` is called at the end of both `createLiveOSIsoImage` and `createImageFromUnchangedOS`. Would it make sense to move the `populatePXEArtifactsDir` call to `convertWriteableFormatToOutputImage` instead?

Reply:
Experimented with moving it and the problem is that we'd have to bubble up the merged configs across multiple functions - and where such state does not make a lot of sense to be returned from those functions.

We need the mergedConfigs because it holds the dracut version in the rootfs - from the current run, or from the previous run if the input is iso.

It will look something like this:
```
{
  if from non-iso
  - mergedConfigs = createLiveOSIsoImage
     - mergedConfigs = prepareArtifactsFromFullImage
       -  mergedConfigs = prepareLiveOSDir
         - mergedConfigs = mergeConfigs()
  else if from iso
  - mergedConfigs = createImageFromUnchangedOS
     - mergedConfigs = mergeConfigs()

  If mergedConfigs.dracutVersion is ok, export folder.
}
```

I think it's more safer and cleaner to leave mergedConfigs contained and not bubble in these new functions. So, I'll leave it as it is.